discovery/eureka/metadata_marshaller.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eureka
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"regexp"
+)
+
+type MetaData struct {
+	Map   map[string]string
+	Class string
+}
+
+type Vraw struct {
+	Content []byte `xml:",innerxml"`
+	Class   string `xml:"class,attr" json:"@class"`
+}
+
+func (s *MetaData) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
Why would we need to marshal to xml?

Reply:
We don't need a marshal. I will delete it